corporate/views/remote_billing_page.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import Any, Dict, Optional, TypedDict
+
+from django.conf import settings
+from django.core import signing
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import csrf_exempt
+
+from zerver.lib.exceptions import JsonableError, MissingRemoteRealmError
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.validator import check_bool, check_dict_only, check_float, check_int, check_string
+from zilencer.models import RemoteRealm, RemoteZulipServer
+
+billing_logger = logging.getLogger("corporate.stripe")
+
+
+SIGNED_ACCESS_TOKEN_VALIDITY_IN_SECONDS = 60 * 60 * 24
+
+
+class RemoteBillingIdentityDict(TypedDict):
+    user_uuid: str
+    user_email: str
+    user_full_name: str
+    remote_server_uuid: str
+    remote_realm_uuid: Optional[str]
+
+
+@csrf_exempt
+@has_request_variables
+def remote_server_billing_entry(
+    request: HttpRequest,
+    remote_server: RemoteZulipServer,
+    user_uuid: str = REQ(json_validator=check_string),
+    user_email: str = REQ(json_validator=check_string),
+    user_full_name: str = REQ(json_validator=check_string),
+    realm_info: Dict[str, Any] = REQ(
+        json_validator=check_dict_only(
+            required_keys=[
+                ("id", check_int),
+                ("host", check_string),
+                ("url", check_string),
+                ("deactivated", check_bool),
+                ("date_created", check_float),
+            ],
+            # Servers before 8.0 don't have these keys.
+            optional_keys=[
+                ("uuid", check_string),
+                ("uuid_owner_secret", check_string),
+            ],
+        )
+    ),
```

Comment:
This might be cleaner with `@typed_endpoint`, since then we can declare the type of the `realm_info` once and then have both this endpoint and `remote_server_post_analytics` process it into a shared data class. 

Though I think that only makes sense if my thought in https://github.com/zulip/zulip/pull/27575/commits/b820be56926e79e01de3560917e50134a2e33e8a#r1395066450 that this code path will be for 8.0+ only is correct, and thus we don't need a different type here.

Reply:
Okay, I guess that necessitates changing `remote_server_post_analytics` to be a `@typed_endpoint` too, right? I pushed a prep commit to do that - and then have this new `remote_server_billing_entry` re-use the `RealmDataForAnalytics` structure in the prototype commit, as a `@typed_endpoint` as well

Reply:
Yeah this looks pretty clean. I guess we can also push some of the typing down into `validate_incoming_table_data` as a follow-up, but that's independent of the rest of what we're doing.